File: 7 StacksandQues/Nesting/nesting.js
function solution(S) {
  let count = 0;
  for (let i = 0; i < S.length; i++) {
    if (S[i] == "(") count++;
    else if (S[i] == ")") count--;
    if (count < 0) return 0
  }
  return count === 0 ? 1 : 0
}

//console.log(solution("(()(())())")); // 1

File: 7 StacksandQues/Nesting/nesting.cpp
int solution(string &S) {
  int count = 0;
  for (int i = 0; i < S.size(); i++) {
    //cout << S.at(i);
    if (S.at(i) == '(')
      count++;
    else if (S.at(i) == ')')
      count--;
    if (count < 0) return 0;
  }
  return count == 0 ? 1 : 0;
}
